components/ILIAS/Chatroom/Chatroom.php

```diff
@@ -32,15 +32,20 @@ public function init(
         array | \ArrayAccess &$pull,
         array | \ArrayAccess &$internal,
     ): void {
-        $contribute[\ILIAS\Setup\Agent::class] = static fn() =>
-            new \ilChatroomSetupAgent(
-                $pull[\ILIAS\Refinery\Factory::class]
-            );
-        $contribute[Component\Resource\PublicAsset::class] = fn() =>
-            new Component\Resource\ComponentJS($this, "chatroom.js");
-        $contribute[Component\Resource\PublicAsset::class] = fn() =>
-            new Component\Resource\ComponentJS($this, "iliaschat.jquery.js");
-        $contribute[Component\Resource\PublicAsset::class] = fn() =>
-            new Component\Resource\ComponentCSS($this, "chatroom.css");
+        $contribute[\ILIAS\Setup\Agent::class] = static fn() => new \ilChatroomSetupAgent($pull[\ILIAS\Refinery\Factory::class]);
+
+        $files = [
+            '../chat/node_modules/socket.io-client/dist/socket.io.min.js',
+            'js/dist/Chatroom.min.js',
+        ];
+
+        $type = ['js' => Component\Resource\ComponentJS::class, 'css' => Component\Resource\ComponentCSS::class];
+
+        foreach ($files as $file) {
+            $class = $type[substr($file, strrpos($file, '.') + 1)];
+            if (file_exists(__DIR__ . '/resources/' . $file)) {
```

Comment:
Do we really need the `file_exists` here?

Reply:
Yes we do. The `socket.io.min.js` file exists only when npm install has been run in `compontents/ILIAS/Chatroom`. Without the `file_exists` check even installations where no chat server is used the `composer du*` step would fail because it could not copy `socket.io.min.js`.